### PR TITLE
Add Signer Standards Overview

### PR DESCRIPTION
**Pull request by frederikrothenberger**
This PR adds an overview document to put in context the various ICRCs drafts that are being worked on in this working group. It also shows how the standards relate to each other and aims to explain (some of) the design decisions behind them.

Note: The document is very likely not complete yet and probably should be expanded in the future. But this PR should establish a basis to iterate on.

### CODE REVIEW

**Comment by frederikrothenberger** [comment]
[Rendered version](https://github.com/dfinity/wg-identity-authentication/blob/frederik/signer-standards/topics/signer_standards_overview.md)

File: topics/signer_standards_overview.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    S ->> C: Request consent message
+    C ->> S: Consent message response
+    S ->> U: Display consent message<br>(Transaction details)
+    U ->> S: Approve request
+    S ->> C: Submit canister call
+    S ->> S: Poll for canister call result
+    S ->> U: Display success message
+    S ->> RP: Canister call response
+```
+This diagram has been simplified, see [ICRC-25](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_25_signer_interaction_standard.md) and [ICRC-21](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_21_consent_msg.md) for more details.
```
Comment (przydatek) [inline topics/signer_standards_overview.md:125]: nit: maybe incorporate also this remark into the sentence before the diagram?

File: topics/signer_standards_overview.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+## Signer Architecture
+
+The signer architecture is not part of the standards. Any component that fulfills the following requirements can act as a signer:
+- Ability to sign messages with an IC supported signature scheme
+- Ability to verify messages signed with an IC supported signature scheme
+- Ability to interact with the IC to send messages and receive responses
+- Implementation of an appropriate transport channel to communicate with a relying party
+
+User interaction is not mandated as developers should be able to build non-interactive signers. However, most common
```
Comment (przydatek) [inline topics/signer_standards_overview.md:135]: maybe replace "to build non-interactive signers" with something like "to build signers that don't interact directly with users" (a singer that offers the required API mentioned above is not "non-interactive")

File: topics/signer_standards_overview.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+This document gives an overview over the ICRC standards related to signer interactions and transaction approval flows.
+It explains the motivation behind the standards and how they relate to each other.
+
+The following ICRC standards are relevant in this context:
+* The [Internet Computer Interface Specification](https://internetcomputer.org/docs/current/references/ic-interface-spec/), specifically the [HTTPs interface](https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-interface)
+* [ICRC-21](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_21_consent_msg.md): Canister Call Consent Messages
+* [ICRC-25](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_25_signer_interaction_standard.md): Signer Interaction Standard
+* ICRC-27: ICRC-25 Extension for ICRC-1 ledger subaccounts
+* ICRC-29: PostMessage Transport Standard for ICRC-25 (still to write, does not yet exist)
```
Comment (przydatek) [inline topics/signer_standards_overview.md:11]: nit: "still to write" -> "still to be written"

File: topics/signer_standards_overview.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+The signer architecture is not part of the standards. Any component that fulfills the following requirements can act as a signer:
+- Ability to sign messages with an IC supported signature scheme
+- Ability to verify messages signed with an IC supported signature scheme
+- Ability to interact with the IC to send messages and receive responses
+- Implementation of an appropriate transport channel to communicate with a relying party
+
+User interaction is not mandated as developers should be able to build non-interactive signers. However, most common
+implementation  is expected to be a signer that is interactive and displays transaction information to the user and 
+receives user input synchronously.
+
```
Comment (przydatek) [inline topics/signer_standards_overview.md:138]: maybe add a sub-section like 
"### Example types of compliant signers" that contains all examples below?

**Review by przydatek — COMMENTED** [review]
Looks like a good starts, please see some questions/suggestions below (mostly nit-picking ;-)

**Comment by frederikrothenberger** [comment]
> Looks like a good starts, please see some questions/suggestions below (mostly nit-picking ;-)

Thanks for the many comments. I tried to address all of them. 🙂

File: topics/signer_standards_overview.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+can interact with each other, _without_ having to specifically integrate with any particular component (or even know
+about each others existence prior to the interaction). To help with this, services maintained by DFINITY will be upgraded
+to support these standards, when they are adopted.
+
+## Assumptions
+* The signer is trusted by the user.
+* The target canister is trusted by the user. Interactions with malicious canisters are not covered by the standards. In particular, interacting with a malicious canister can produce arbitrary outcomes regardless of how the calls were signed.
+* The relying party is _not_ trusted by the user nor the signer and might be malicious.
+* The signer is _not_ trusted by the relying party. The relying party must be able to detect malicious behaviour from a signer.
+  This property is important for an open ecosystem where relying parties openly interact with any standard compliant signer.
```
[the remark this answers is not in the document]
Comment (frederikrothenberger) [reply topics/signer_standards_overview.md]: I added more explanation and an example. PTAL 🙂

**Review by przydatek — APPROVED** [review]
LGTM, thanks!

**Pull request merged**